### PR TITLE
Feat(web): Introduce `FileUploader` component #DS-664

### PR DESCRIPTION
**Pull request by adamkudrna**
obrazek

### CODE REVIEW

File: packages/web/src/scss/components/FileUploader/index.html [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+        <li>
+
+          <!-- FileUploaderAttachment: start -->
+          <div class="FileUploaderAttachment">
+            <svg width="24" height="24" aria-hidden="true">
+              <use xlink:href="/icons/svg/sprite.svg#file" />
+            </svg>
+            <div class="text-truncate">My resume.docx</div>
+            <button type="button" class="FileUploaderAttachment__remove">
+              <span class="accessibility-hidden">Delete</span>
```
Comment (crishpeen) [inline packages/web/src/scss/components/FileUploader/index.html:189]: Remove?

File: packages/web/src/scss/components/FileUploader/index.html [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+</section>
+<section class="docs-Section">
+
+  <h2 class="docs-Heading">Fluid Width</h2>
+
+  <!-- FileUploader: start -->
+  <div class="FileUploader FileUploader--fluid">
+
+    <!-- FileUploaderInput: start -->
+    <div class="FileUploaderInput FileUploaderInput--fluid has-drag-and-drop">
```
Comment (crishpeen) [inline packages/web/src/scss/components/FileUploader/index.html:279]: `FileUploaderInput--fluid` do we need this class?

File: packages/web/src/scss/components/FileUploader/index.html [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+          <span class="FileUploaderInput__dragAndDropLabel">or drag and drop here</span>
+        </label>
+        <div class="FileUploaderInput__helperText">Max file size is 10 MB</div>
+        <input type="file" id="fileUploadFluid" class="FileUploaderInput__input" />
+      </div>
+    </div>
+    <!-- FileUploaderInput: end -->
+
+    <ul class="FileUploader__list">
+      <li>
```
Comment (crishpeen) [inline packages/web/src/scss/components/FileUploader/index.html:296]: aria-labelledby? :)

File: packages/web/src/scss/components/FileUploader/index.html (hunk 1 above)
Comment (adamkudrna) [reply packages/web/src/scss/components/FileUploader/index.html:189]: Probably yes…

File: packages/web/src/scss/components/FileUploader/index.html (hunk 2 above)
Comment (adamkudrna) [reply packages/web/src/scss/components/FileUploader/index.html:279]: No, good point.

File: packages/web/src/scss/tools/_form-fields.scss [hunk 4]
```diff
@@ -101,20 +101,28 @@
...
     @each $variant-name, $variant-value in form-fields-theme.$box-field-variants {
-        :is(.#{$field-name}--#{$variant-name}, .#{$field-name}.has-#{$variant-name}) > .#{$field-name}__input {
-            border-color: map.get($variant-value, input-border-color);
+        @if $field-name == 'FileUploaderInput' {
+            :is(.#{$field-name}--#{$variant-name}, .#{$field-name}.has-#{$variant-name}) > .#{$field-name}__dropZone {
+                border-color: map.get($variant-value, input-border-color);
+            }
+        } @else {
+            :is(.#{$field-name}--#{$variant-name}, .#{$field-name}.has-#{$variant-name}) > .#{$field-name}__input {
+                @if $field-name != 'FileUploaderInput' {
```
[the remark this answers is not in the document]
Comment (adamkudrna) [reply packages/web/src/scss/tools/_form-fields.scss:110]: Good point, I forgot it there during the development.

**Review by crishpeen — APPROVED** [review]
👍 💪

File: packages/web/src/scss/components/FileUploader/README.md [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+  <!-- FileUploaderList -->
+</div>
+```
+
+## FileUploaderInput
+
+FileUploaderInput is a file picker built around the native HTML
+[file `input`][mdn-input-file].
+
+If supported by the device, FileUploaderInput automatically turns on the
```
[the remark this answers is not in the document]
Comment (adamkudrna) [reply packages/web/src/scss/components/FileUploader/README.md:57]: Yes. I put it there in advance because I expect us to forget 🙈.

**Review by literat — APPROVED** [review]
LGTM

**Review by tomassychra — APPROVED** [review]

**Pull request merged**